app/models/post.rb

```diff
@@ -1,4 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :user
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def take_five_recent
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_user_post_counter
+    author.update(post_counter: author.posts.count)
+  end
```

Comment:
## Optional
- [ ] I highly suggest that you use [Rails active callback](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the `update_user_post_counter` method every time a new post has been saved to the database, then you make `update_user_post_counter` method private